Add docs for 2023-01 decorators (#2715)
babel/website: docs/plugin-proposal-decorators.md

  <summary>History</summary>
| Version | Changes |
| --- | --- |
| `v7.21.0` | Added support for `version: "2023-01"` |
| `v7.19.0` | Added support for `version: "2022-03"` |
| `v7.17.0` | Added the `version` option with support for `"2021-12"`, `"2018-09"` and `"legacy"` |
</details>

### `version`

> `"2023-01"`, `"2022-03"`, `"2021-12"`, `"2018-09"` or `"legacy"`.

Reviewer: Is a `version` a required option now? If so it would be a breaking change.

Author: No, just `["@babel/plugin-proposal-decorators"]` has always been invalid. `"version"` has always been required, except if `"decoratorsBeforeExports"` was set. Since most people will not need to set it now, it's confusing to say that `"version"` has a default value (however, this behavior is still mentioned a few lines below).

export class Foo {}
```

> This option was added to help tc39 collect feedback from the community by allowing experimentation with both possible syntaxes. The proposal now settled on allowing decorators either before or after `export`.

Reviewer 2: ```suggestion
This option was originally added to help tc39 collect feedback from the community by allowing experimentation with the proposed syntaxes. The proposal has now settled on allowing decorators either before or after `export`.
```
